cookbook/docs/integrations.rst

```diff
@@ -1,48 +1,60 @@
+The data is automatically marshalled and unmarshalled into and out of the plugin. Users should implement :py:class:`flytekit.core.base_task.PythonTask` API, defined in Flytekit.
 
-Flytekit Plugins are lazily loaded and can be released independently like libraries. We follow a convention to name the plugin like
-``flytekitplugins-*``, where * implies the capability. For example ``flytekitplugins-papermill`` enables users to author flytekit tasks using `Papermill <https://papermill.readthedocs.io/en/latest/>`_
+Implementation 
+^^^^^^^^^^^^^^^
 
-Examples of flytekit only plugins:
+Flytekit Plugins are simply loaded and can be released independently like libraries. We follow a convention to name the plugin like
+``flytekitplugins-*``, where * implies the capability. 
+
+For example ``flytekitplugins-papermill`` enables users to author flytekit tasks using `Papermill <https://papermill.readthedocs.io/en/latest/>`_.
+
+Examples of Flytekit-Only Plugins
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 #. Papermill implementation `flytekitplugins-papermill <https://github.com/lyft/flytekit/tree/master/plugins/papermill>`_
 #. SQLite3 implementation `SQLite3 Queries <https://github.com/lyft/flytekit/blob/master/flytekit/extras/sqlite3/task.py>`_
 
 What are Backend Plugins?
 =========================
+
+Flyte has a unique capability of adding backend plugins. Backend plugins enable the Flyte platform to add new capabilities. This has several advantages, mainly:
+
+#. Advanced introspection capabilities - ways to improve logging, etc.
+#. Service oriented architecture - ability to bugfix, deploy plugins without releasing new libraries and forcing all users to update their libraries.
+#. Better management of the system communication - for example in case of aborts, Flyte can guarantee cleanup of the remote resources.
+#. Reduced cost overhead, for many plugins which launch jobs on a remote service or cluster, the plugins are essentially just polling. This has a huge compute cost in traditional architectures like Airflow etc. Flyte on the other hand, can run these operations in its own control plane.
+#. The potential to create drastically new interfaces that work across multiple languages and platforms.
+
 Flyte backend plugins are more involved and implementation needs writing code in ``Golang`` that gets plugged into the Flyte backend engine. These plugins are statically loaded into the FlytePropeller. The contract for the plugin can be encoded in any serialization format - e.g. JSON, OpenAPI, protobuf. The community in general prefers using protobuf.
 Once the backend plugin is implemented, any language SDK can be implemented to provide a specialized interface for the user.
 
 Examples
-
+^^^^^^^^^
 #. `Sagemaker <https://github.com/lyft/flytekit/tree/master/plugins/awssagemaker>`_
 #. `K8s Spark <https://github.com/lyft/flytekit/tree/master/plugins/spark>`_
 
```

Comment:
Do we still need these examples here?

Reply:
I don't think so, we can delete the examples

Reply:
Do we want to move the links to the examples below? Or keep them as is without links?

Reply:
Removed the examples as they can be found in the integrations TOC hierarchy.

Reply:
Thanks Samhita!